src/menu/menu.c

```diff
@@ -115,6 +124,10 @@ static void menu_init (boot_params_t *boot_params) {
         __boot_tvtype = TV_NTSC;
     }
 
+    if (!menu->settings.sound_enabled) {
+        mixer_ch_set_vol(SFX_CHANNEL, 0, 0);
+    }
```

Comment:
This still causes libdragon mixer to play sound, even if it can't be heard. Better solution would be checking this flag in `sound_play_effect` and avoid calling `wav64_play` when sounds are disabled.